staging/src/k8s.io/apiserver/pkg/endpoints/discovery/legacy.go

```diff
@@ -66,11 +70,76 @@ func (s *legacyRootAPIHandler) WebService() *restful.WebService {
 }
 
 func (s *legacyRootAPIHandler) handle(req *restful.Request, resp *restful.Response) {
+	hash, err := getGroupVersionHash("/api/v1", s.handler)
+	if err != nil {
+		panic(err)
+	}
 	clientIP := utilnet.GetClientIP(req.Request)
 	apiVersions := &metav1.APIVersions{
 		ServerAddressByClientCIDRs: s.addresses.ServerAddressByClientCIDRs(clientIP),
 		Versions:                   []string{"v1"},
+		Hashes:                     map[string]string{"v1": hash},
 	}
 
 	responsewriters.WriteObjectNegotiated(s.serializer, negotiation.DefaultEndpointRestrictions, schema.GroupVersion{}, resp.ResponseWriter, req.Request, http.StatusOK, apiVersions)
 }
+
+func getGroupVersionHash(path string, handler http.Handler) (string, error) {
```

Comment:
Jeffrey added a copy of `GetGroupVersionHash`/`inMemoryResponseWriter` to https://github.com/Jefftree/kubernetes/blob/feature-discovery-cache-busting/staging/src/k8s.io/apiserver/pkg/endpoints/discovery/etag.go

Should remove these and use that shared version instead